src/commands/generate/fromTemplate.ts

```diff
@@ -213,14 +228,30 @@ export default class Template extends Command {
     return { asyncapi, template, output };
   }
 
-  private parseFlags(disableHooks?: string[], params?: string[], mapBaseUrl?: string): ParsedFlags {
+  private parseFlags(disableHooks?: string[], params?: string[], mapBaseUrl?: string, registryUrl?: string, registryAuth?:string, registryToken?:string): ParsedFlags {
     return {
       params: this.paramParser(params),
       disableHooks: this.disableHooksParser(disableHooks),
       mapBaseUrlToFolder: this.mapBaseURLParser(mapBaseUrl),
+      registryURLValidation: this.registryURLParser(registryUrl),
+      registryAuthentication: this.registryValidation(registryUrl, registryAuth, registryToken)
+      
     } as ParsedFlags;
   }
 
+  private registryURLParser(input?:string) {
+    if (!input) { return; }
+    const isURL = /^https?:/;
+    if (!isURL.test(input.toLowerCase())) {
+      throw new Error('Invalid --registry-url flag. The param requires a valid http/https url.');
+    }
+  }
+  private async registryValidation(registryUrl?:string, registryAuth?:string, registryToken?:string) {
+    const response= await axios.get(registryUrl as string);
+    if (response.status === 400&&!registryAuth&&!registryToken) {
```

Comment:
I believe auth error results in `401` error code.

Reply:
Done @Shurtu-gal